azure-cli/plan.sh

```diff
@@ -0,0 +1,59 @@
+pkg_name=azure-cli
+pkg_origin=core
+pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
+pkg_license=('MIT')
+pkg_description="The Azure command-line interface (CLI) is Microsoft's cross-platform \
+  command-line experience for managing Azure resources. Use it in your browser with Azure \
+  Cloud Shell, or install it on macOS, Linux, or Windows and run it from the command line."
+pkg_upstream_url=https://docs.microsoft.com/en-us/cli/azure
+pkg_version=2.0.57
+pkg_shasum="0777d6c301bf741a2fdf9d2053a3a9285a4e0f5a067b64659e77c1ca9ed71e7a"
+pkg_source="https://github.com/Azure/azure-cli/archive/azure-cli-${pkg_version}.tar.gz"
+pkg_dirname=${pkg_name}-${pkg_name}-${pkg_version}
+pkg_build_deps=(
+  core/gcc
+  core/linux-headers
+  core/util-linux
+  core/musl
```

Comment:
Is there a reason for `core/musl` in the build deps?  

I think we want to remove this and add `core/glibc` to `pkg_deps` instead.  Looking at the ldd output for .so files in the package, they link against libpthread and libc from glibc.

```
ldd /hab/pkgs/smacfarlane/azure-cli/2.0.57/20190208181333/lib/python3.7/site-packages/cryptography/hazmat/bindings/_constant_time.abi3.so
	linux-vdso.so.1 (0x00007ffe7bfb3000)
	libpthread.so.0 => /hab/pkgs/core/glibc/2.27/20190115002733/lib/libpthread.so.0 (0x00007f6050803000)
	libc.so.6 => /hab/pkgs/core/glibc/2.27/20190115002733/lib/libc.so.6 (0x00007f605064b000)
```

Reply:
I used `core/musl` since that is what they installed in the docker image I referenced: https://github.com/Azure/azure-cli/blob/dev/Dockerfile#L37

I'm not against using `core/glibc` though.